From c04efaaf24f5de68d6c669f3197e9ff12d8a9dbc Mon Sep 17 00:00:00 2001
From: Yan Markman <ymarkman@marvell.com>
Date: Tue, 27 Nov 2018 10:45:27 +0200
Subject: [PATCH 0811/1051] net: mvpp2: delete mtu change alignment check

This patch does not change any functionality but is cleanup.

Delete mtu-to-packetSize un-alignment checker from change-mtu
procedure since the packetSize obtained from mtu already/always
aligned onto cache-line-size and therefore the if() is always false.

Change-Id: I90c950186df267cdc3b2a6082776629c4abd3bc5
Signed-off-by: Yan Markman <ymarkman@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61347
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
index 0d6b7cf9ef62..53cc9537dee2 100644
--- a/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2/mvpp2_main.c
@@ -4444,12 +4444,6 @@ static int mvpp2_change_mtu(struct net_device *dev, int mtu)
 		return -EPERM;
 	}
 
-	if (!IS_ALIGNED(MVPP2_RX_PKT_SIZE(mtu), 8)) {
-		netdev_info(dev, "illegal MTU value %d, round to %d\n", mtu,
-			    ALIGN(MVPP2_RX_PKT_SIZE(mtu), 8));
-		mtu = ALIGN(MVPP2_RX_PKT_SIZE(mtu), 8);
-	}
-
 	if (!netif_running(dev)) {
 		err = mvpp2_bm_update_mtu(dev, mtu);
 		if (!err) {
-- 
2.17.1

